framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -469,7 +469,7 @@ DerivativeMaterialInterface<T>::validateCoupling(const MaterialPropertyName & ba
 
   // iterate over all variables in the both the non-linear and auxiliary system (optional)
   validateCouplingHelper<U, is_ad>(
-      prop_name, c, _dmi_fe_problem.getNonlinearSystemBase().system(), missing);
+      prop_name, c, _dmi_fe_problem.getNonlinearSystemBase(/*nl_sys=*/0).system(), missing);
```

Comment:
Is tehre a way to get the proper system from `T` (which would be one of `Material`, `Kernel`, `AuxKernel` etc.)

Reply:
The only way I can think of right now is through these parameters of a MooseObject:
```
  params.addPrivateParam<SystemBase *>("_sys", nullptr);
  params.addPrivateParam<SystemBase *>("_nl_sys", nullptr);
```
I am not sure if these are set for every object type. But if it isn't you can default to 0, I suppose.

Reply:
> Is there a way to get the proper system from T (which would be one of Material, Kernel, AuxKernel etc.)?

Often-times there is, however, with materials there is not. Materials are system agnostic, e.g. they can compute quantities from variables in all systems